goroutine/gid_test.go

```diff
@@ -35,8 +39,15 @@ func TestID(t *testing.T) {
 	wg.Wait()
 }
 
+var testID int64
+
 // goid returns the goroutine ID extracted from a stack trace.
 func goid() int64 {
+	if IsTinyGo {
+		// pretty horrible test that aligns with the implementation, but at least it tests we get 1,2,3... different numbers.
+		testID++
+		return testID
+	}
```

Comment:
Shouldn't use atomic.Int64 ?

Reply:
for the same reason there is a mutex even though it's not needed, possibly, yet this is test code that we know is actually not run in parallel

Reply:
did